packages/ember-core/src/components/form/file-upload.gts

```diff
@@ -0,0 +1,269 @@
+import { registerDestructor } from '@ember/destroyable';
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import { htmlSafe } from '@ember/template';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { t } from 'ember-intl';
+
+import BoundValue from './bound-value.ts';
+import { classes } from '../../helpers/classes.ts';
+import onInsert from '../../modifiers/on-insert.ts';
+import onUpdate from '../../modifiers/on-update.ts';
+import { FileValidator } from '../../validation/index.ts';
+import Button from '../button.gts';
+
+import type { FormType } from './index.gts';
+
+export interface SelectedFileListSignature {
+  Args: {
+    files: File[] | null;
+    isInvalid?: boolean;
+    isWarning?: boolean;
+    onRemove?: (index: number) => void;
+  };
+  Element: HTMLUListElement;
+  Blocks: {
+    default: [];
+  };
+}
+
+export interface FileUploadSignature {
+  Args: {
+    accept?: string[];
+    describedBy?: string;
+    disabled?: boolean;
+    form?: FormType;
+    id?: string;
+    isInvalid?: boolean;
+    isWarning?: boolean;
+    validatorKey?: string;
+    onAdd?: (files: File[]) => unknown;
+    onRemove?: (file: File) => unknown;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLDivElement;
+}
+
+class SelectedFileList extends Component<SelectedFileListSignature> {
+  @action
+  removeFile(index: number) {
+    this.args.onRemove?.(index);
+  }
+
+  <template>
+    {{#if @files}}
+      <div class="m-0 p-0 col-12">
+        <ul class={{classes "list-group list-group-flush rounded p-0 col-12 form-control" (if @isInvalid "is-invalid") (if @isWarning "is-warning")}}>
+          {{#each @files as |file index|}}
+            <li class="col-12 list-group-item d-flex flex-row align-items-center justify-content-between">
+              {{file.name}}
+              <Button data-test-remove class="btn-link" @onClick={{fn this.removeFile index}}>{{t "nrg.file-upload.remove"}}</Button>
```

Comment:
I would recommend hiding the remove button if the input is disabled.

Reply:
To be safe, I would also make sure the input isn't disabled before running `change` or `removeFile`.